docs/articles/documentation/extending-testcafe/creating-custom-reporter/reporterpluginhost.md

```diff
@@ -0,0 +1,280 @@
+---
+layout: docs
+title: ReporterPluginHost Object
+permalink: /documentation/extending-testcafe/creating-custom-reporter/reporterpluginhost/
+---
+# ReporterPluginHost Object
+
+The ReporterPluginHost object provides access to a custom reporter scaffolded by the [TestCafe reporter generator](https://github.com/DevExpress/generator-testcafe-reporter).
+
+The object has some main methods that you need to implement and a number of helper methods.
+
+## Methods to Implement
+
+To implement your custom reporter, you need to implement several methods.
+
+The following four methods should output the desired information at certain moments during test run.
+All the required data is provided to these methods through their parameters.
+
+To output information, use [helper methods](#helper-methods) and libraries exposed by the object.
+
+The object you are currently editing will be mixed in with `ReporterPluginHost`, so you can refer to its methods by using `this`.
+
+### reportTaskStart
+
+Fires when a test task is started, that is at the beginning of testing.
+
+```js
+reportTaskStart (/* startTime, userAgents, testCount */) {
+    //To implement
+}
+```
+
+**Example**
+
+```js
+reportTaskStart (startTime, userAgents, testCount) {
```

Comment:
We need some really simple example. E.g. output data in some plain text format: `{passed,failed}-fixture-test`, then `passed/failed - duration: {duration}`. And build example on top of it.